### ISSUE TITLE
`npx inlang lint` creates empty `undefined.json` files

### ISSUE THREAD

**Issue by NiklasBuchfink** [inlang/inlang#966]
## Problem

Executing `npx inlang lint` creates an empty `undefined.json` file in each language folder with the content:

```js
{}
```

**Additional information:**
- pathPattern: `public/locales/{language}/*.json` (folder structure with * structure)
- Operating system: Windows 11

<img width="704" alt="image" src="https://github.com/inlang/inlang/assets/59048346/67815c8f-4b45-4e12-9546-7ce8d38d58fa">

### Expected behavior

When running the command, no empty json file should be created.

## Steps to reproduce

1. Setup `@inlang/cli` with a i18next project
2. Make shure it's a project with a pathPattern like `public/locales/{language}/*.json`
3. Execute `npx inlang lint`

**Comment by felixhaeberle** [comment]
@NiklasBuchfink Thanks for submitting this bug!

**Comment by felixhaeberle** [comment]
@samuelstroschein Does lint write at all? 😅 Maybe that's only a bug inside the CLI but most of the time, the CLI only uses functions exposed by inlang/core.

**Comment by samuelstroschein** [comment]
@felixhaeberle `setupConfig` writes to the filesystem to make a roundtrip test. 

- we should probably use a memory fs for roundtrip testing
- something is wrong in a plugin then i guess

**Comment by felixhaeberle** [comment]
@NiklasBuchfink Can you provide the **config file** and the output of `npx @inlang/cli config validate` for reproduction?

**Comment by felixhaeberle** [comment]
Probably related to #976

**Comment by samuelstroschein** [comment]
@felixhaeberle true, could be related to #976 

I'll investigate today

**Comment by NiklasBuchfink** [comment]
@felixhaeberle 

> @NiklasBuchfink Can you provide the **config file** and the output of `npx @inlang/cli config validate` for reproduction?

Example repo: https://github.com/NiklasBuchfink/nextjs-i18next

**Config:**
```js
/**
 * @type { import("@inlang/core/config").DefineConfig }
 */
export async function defineConfig(env) {
  const { default: pluginJson } = await env.$import(
    "https://cdn.jsdelivr.net/npm/@inlang/plugin-json@3/dist/index.js"
  )

  const { default: standardLintRules } = await env.$import(
    "https://cdn.jsdelivr.net/npm/@inlang/plugin-standard-lint-rules@3/dist/index.js",
  );

  return {
    referenceLanguage: "en",
    plugins: [
      pluginJson({ pathPattern: "./app/i18n/locales/{language}/*.json" }),
      standardLintRules(),
    ],
  };
}
```

Output `npx @inlang/cli config validate`:

<img width="438" alt="image" src="https://github.com/inlang/inlang/assets/59048346/6b8841cc-fd05-4be4-991f-24f8894152f7">

Seems to be related to this as existing files are not recognized and new ones are created.

**Comment by NiklasBuchfink** [comment]
Closed with #987
